### PR TITLE
v13 release blog

### PR DESCRIPTION
**Pull request by alexagriffith**
"Fixes #issue-number" or "Add description of the problem this PR solves"
Release blog for KServe version 0.13
## Proposed Changes 

-
-
-

### CODE REVIEW

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+### vLLM support 
+
+Version 0.13 introduces dedicated runtime support for [vLLM](https://docs.vllm.ai/en/latest/), for enhanced transformer model serving. This support now includes auto-mapping vLLMs as the backend for supported tasks, streamlining the deployment process and optimizing performance.
+
+
+### OpenAI Schema Integration
+
+Embracing the OpenAI protocol, KServe v0.13 now supports three specific endpoints:
```
Comment (terrytangyuan) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:44]: Perhaps clarify that these are additional endpoints for special purposes

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+* `/v1/completions`
+* `/v1/chat/completions`
+* `/v1/models`
+
+This update fosters a standardized approach to transformer model serving, ensuring compatibility with a broader spectrum of models and tools, and enhances the platform's versatility.
+
+
+## Key Changes
+
+This release includes several fixes key issues and added features:
```
Comment (terrytangyuan) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:55]: Link to a complete list and mention that this list is not exhaustive

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+- Visit our [Website](https://kserve.github.io/website/) or [GitHub](https://github.com/kserve)
+- Join the Slack ([#kserve](https://kubeflow.slack.com/?redir=%2Farchives%2FCH6E58LNP))
+- Attend our community meeting by subscribing to the [KServe calendar](https://wiki.lfaidata.foundation/display/kserve/calendars).
+- View our [community github repository](https://github.com/kserve/community) to learn how to make contributions. We are excited to work with you to make KServe better and promote its adoption!
+
+For complete details on the new features and updates, visit our [official release notes](https://github.com/kserve/kserve/releases/tag/v0.13.0-rc0).
+
+Thanks for all the contributors who have made the commits to 0.13 release!
+
+The KServe Working Group
```
Comment (terrytangyuan) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:75]: KServe project

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# From Serverless Predictive Inference to Generative Inference: Introducing KServe v0.13
+
+We are excited to unveil KServe v0.13, marking a significant leap forward in providing features for the evolving demands of model serving. This release is highlighted by three pivotal updates: enhanced Huggingface runtime, robust vLLM support, and the integration of OpenAI protocol standards.
+
+Here is a summary of the key changes:
```
Comment (terrytangyuan) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:5]: End the sentence before another section

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 3 above)
Comment (yuzisun) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md:75]: Also nice to acknowledge the major contributors for this release :)

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+          memory: 24Gi
+          nvidia.com/gpu: "1"
+```
+
+### vLLM support 
+
+Version 0.13 introduces dedicated runtime support for [vLLM](https://docs.vllm.ai/en/latest/), for enhanced transformer model serving. This support now includes auto-mapping vLLMs as the backend for supported tasks, streamlining the deployment process and optimizing performance.
+
+
+### OpenAI Schema Integration
```
Comment (yuzisun) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:42]: Good to explain the motivation why KServe Integrates OpenAI Schema.

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+          nvidia.com/gpu: "1"
+        requests:
+          cpu: "6"
+          memory: 24Gi
+          nvidia.com/gpu: "1"
+```
+
+### vLLM support 
+
+Version 0.13 introduces dedicated runtime support for [vLLM](https://docs.vllm.ai/en/latest/), for enhanced transformer model serving. This support now includes auto-mapping vLLMs as the backend for supported tasks, streamlining the deployment process and optimizing performance.
```
Comment (yuzisun) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:39]: Good to clarify that vLLM is one of the backends currently KServe supports and in future KServe plans to support other runtimes like TensorRT-LLM.

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
+# From Serverless Predictive Inference to Generative Inference: Introducing KServe v0.13
+
+We are excited to unveil KServe v0.13, marking a significant leap forward in providing features for the evolving demands of model serving. This release is highlighted by three pivotal updates: enhanced Huggingface runtime, robust vLLM support, and the integration of OpenAI protocol standards.
+
+Here is a summary of the key changes:
+
+## Enhanced HuggingFace Runtime Support
+
+KServe v0.13 enriches its Huggingface runtime and now supports running HuggingFace models out-of-the-box. KServe v0.13 implements a [KServe HuggingFace Serving Runtime](https://github.com/kserve/kserve/tree/master/python/huggingfaceserver), `kserve-huggingfaceserver`. With this implementation, KServe can now automatically infer a [task](https://huggingface.co/tasks) from model architecture. Currently supported tasks include sequence classification, token classification, fill mask, text generation, and text to text generation.
```
Comment (yuzisun) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:9]:
```suggestion
KServe v0.13 enriches its Huggingface runtime and now supports running HuggingFace models out-of-the-box. KServe v0.13 implements a [KServe HuggingFace Serving Runtime](https://github.com/kserve/kserve/tree/master/python/huggingfaceserver), `kserve-huggingfaceserver`. With this implementation, KServe can now automatically infer a [task](https://huggingface.co/tasks) from model architecture and select the optimized serving runtime. Currently supported tasks include sequence classification, token classification, fill mask, text generation, and text to text generation. 
```

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+Version 0.13 introduces dedicated runtime support for [vLLM](https://docs.vllm.ai/en/latest/), for enhanced transformer model serving. This support now includes auto-mapping vLLMs as the backend for supported tasks, streamlining the deployment process and optimizing performance.
+
+
+### OpenAI Schema Integration
+
+Embracing the OpenAI protocol, KServe v0.13 now supports three specific endpoints:
+
+* `/v1/completions`
+* `/v1/chat/completions`
+* `/v1/models`
```
Comment (yuzisun) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:48]:
```suggestion
* `/openai/v1/completions`
* `/openai/v1/chat/completions`
* `/openai/v1/models`
```

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+This update fosters a standardized approach to transformer model serving, ensuring compatibility with a broader spectrum of models and tools, and enhances the platform's versatility.
```
Comment (yuzisun) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:50]:
```suggestion
This update fosters a standardized approach to transformer model serving, ensuring compatibility with a broader spectrum of models and tools, and enhances the platform's versatility. The API can be directly used with OpenAI's client libraries or third-party tools, like LangChain or LlamaIndex
```

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 6 above)
Comment (johnugeorge) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:39]: Also talk about the case off fallbacking to HF if vllm is not available or doesn't support the model architecture

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 3 above)
Comment (alexagriffith) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md:75]: what do you consider "major"?

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 2 above)
Comment (alexagriffith) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md:55]: yes I linked https://github.com/kserve/website/pull/363/files#diff-54e90e2520c008f2d07108f22dd3cddef5c3a5575c912f27f26d0ea76ecd2fcdR71 but will move it up

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 5 above)
Comment (alexagriffith) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md:42]: to keep up with standards of transformer based models?

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 4 above)
Comment (alexagriffith) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md]: fixed

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 2 above)
Comment (alexagriffith) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md:55]: fixed!

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 6 above)
Comment (alexagriffith) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md]: added

Comment (alexagriffith) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md]: added

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 1 above)
Comment (alexagriffith) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md]: added a bit more, does that work?

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 5 above)
Comment (alexagriffith) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md:68]: added a bit more - lmk how it looks

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 3 above)
Comment (terrytangyuan) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md:104]: "project" would make more sense to me. We can also remove this line :-)

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+          cpu: "6"
+          memory: 24Gi
+          nvidia.com/gpu: "1"
+        requests:
+          cpu: "6"
+          memory: 24Gi
+          nvidia.com/gpu: "1"
+```
+
+KServe plans to support more backend options in the future, such as ensorRT-LLM.
```
Comment (johnugeorge) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:66]:
```suggestion
KServe plans to support more backend options in the future, such as tensorRT-LLM. 
```

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
```
Comment (alexagriffith) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:68]:
```suggestion
KServe plans to support more backend options in the future, such as TensorRT-LLM. 
```

File: docs/modelserving/v1beta1/llm/huggingface/README.md [hunk 12]
```diff
@@ -62,6 +65,35 @@ Sample OpenAI Completions request:
...
+```
+!!! success "Expected Output"
+
+  ```{ .bash .no-copy }
+    {"id":"cmpl-87ee252062934e2f8f918dce011e8484","choices":[{"finish_reason":"length","index":0,"message":{"content":"<generated_response>","tool_calls":null,"role":"assistant","function_call":null},"logprobs":null}],"created":1715353461,"model":"llama2","system_fingerprint":null,"object":"chat.completion","usage":{"completion_tokens":30,"prompt_tokens":3,"total_tokens":33}}
+  ```
+
+Sample OpenAI Chat request:
+
+```bash
```
Comment (alexagriffith) [inline docs/modelserving/v1beta1/llm/huggingface/README.md:95]: i was deep in rebasing and im not sure if this turned out right

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+          nvidia.com/gpu: "1"
+        requests:
+          cpu: "6"
+          memory: 24Gi
+          nvidia.com/gpu: "1"
+```
+
+See more details in our updated docs to [Deploy the Llama2 model with Hugging Face LLM Serving Runtime](https://kserve.github.io/website/master/modelserving/v1beta1/llm/huggingface/).
+
+Additionally, if the HuggingFace backend is preferred over vLLM, vLLM auto-mapping can be disabled with the `--backend=huggingface` arg.
```
Comment (alexagriffith) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:75]: @yuzisun is this right or should it be --disable-vllm ?

File: docs/modelserving/v1beta1/llm/huggingface/README.md (hunk 12 above)
Comment (alexagriffith) [reply docs/modelserving/v1beta1/llm/huggingface/README.md:95]: do we want the generic example or the example with france?

File: docs/modelserving/v1beta1/llm/huggingface/README.md [hunk 14]
```diff
@@ -4,8 +4,11 @@ The Hugging Face LLM serving runtime implements a runtime that can serve Hugging
 In this example, we deploy a Llama2 model from Hugging Face by running an `InferenceService` with [Hugging Face Serving runtime](https://github.com/kserve/kserve/tree/master/python/huggingfaceserver). Based on the performance requirement for large language models, KServe chooses to perform the inference using a more optimized inference engine like [vLLM](https://github.com/vllm-project/vllm) for text generation models.
 
 ### Serve the Hugging Face LLM model using vLLM
-KServe Hugging Face runtime by default uses vLLM to serve the LLM models for faster inference, higher throughput than Hugging Face API, implemented with paged attention, continous batching, optmized CUDA kernel. 
-You can still use `--backend=huggingface` in the container args to fall back to perform the inference using Hugging Face API.
+
+KServe Hugging Face runtime by default uses vLLM to serve the LLM models for faster inference and higher throughput than the Hugging Face API, implemented with paged attention, continuous batching and an optimized CUDA kernel. 
+
+You can still use `--disable_vllm` flag to fall back to perform the inference using Hugging Face API.
```
Comment (sivanantha321) [inline docs/modelserving/v1beta1/llm/huggingface/README.md:10]: This flag is no longer available. We should use --backend=huggingface

File: docs/modelserving/v1beta1/llm/huggingface/README.md (hunk 12 above)
Comment (yuzisun) [reply docs/modelserving/v1beta1/llm/huggingface/README.md:95]: I think this was due to the bad rebase, I do not think you changed this file.

File: docs/modelserving/v1beta1/llm/huggingface/README.md (hunk 14 above)
Comment (alexagriffith) [reply docs/modelserving/v1beta1/llm/huggingface/README.md:10]: ok thanks i think this got messed up in the rebase

**Comment by alexagriffith** [comment]
@theofpa @xfu83 @johnugeorge @sivanantha321 @terrytangyuan can I get a re-review? i had to squash a lot of commits and the history was kind of messed up so I want to make sure I didn't miss anything or leave anything out

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+* KEDA integration for token based LLM Autoscaling [#3561](https://github.com/kserve/kserve/issues/3561).
+
+
+## Other Changes
+
+This release also includes several enhancements and changes:
+
+### What's New?
+* Async streaming support for v1 endpoints [#3402](https://github.com/kserve/kserve/issues/3402).
+* Support for .json and .ubj model formats in the XGBoost server image [#3546](https://github.com/kserve/kserve/issues/3546).
```
Comment (terrytangyuan) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:103]:
```suggestion
* Support for `.json` and `.ubj` model formats in the XGBoost server image [#3546](https://github.com/kserve/kserve/issues/3546).
```

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+* Enhanced flexibility in KServe by allowing the configuration of multiple domains for an inference service [#2747](https://github.com/kserve/kserve/issues/2747).
+* Enhanced the manager setup to dynamically adapt based on available CRDs, improving operational flexibility and reliability across different deployment environments [#3470](https://github.com/kserve/kserve/issues/3470).
+
+### What's Changed?
+* Remove Seldon Alibi dependencie [#3380](https://github.com/kserve/kserve/issues/3380).
```
Comment (terrytangyuan) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:108]:
```suggestion
* Removed Seldon Alibi dependency [#3380](https://github.com/kserve/kserve/issues/3380).
```

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+* Removal of conversion webhook from manifests. [#3344](https://github.com/kserve/kserve/issues/3344).
+
+For complete details on the new features and updates, visit our [official release notes](https://github.com/kserve/kserve/releases/tag/v0.13.0-rc0).
+
+
+## Join the community
+
+- Visit our [Website](https://kserve.github.io/website/) or [GitHub](https://github.com/kserve)
+- Join the Slack ([#kserve](https://kubeflow.slack.com/?redir=%2Farchives%2FCH6E58LNP))
```
Comment (terrytangyuan) [inline docs/blog/articles/2024-05-15-Kserve-0.13-release.md:117]: Let's use this link that uses the CNCF Slack instead: https://github.com/kserve/community?tab=readme-ov-file#questions-and-issues

**Comment by terrytangyuan** [comment]
@alexagriffith Left some minor suggestions. Otherwise LGTM. Great work!

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 13 above)
Comment (gavrishp) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md:75]: This is correct. We have moved away from `--disable-vllm` to use `--backend`

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md (hunk 17 above)
Comment (alexagriffith) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md]: done

File: docs/blog/articles/2024-05-15-Kserve-0.13-release.md [hunk 18]
```diff
@@ -0,0 +1,123 @@
+# From Serverless Predictive Inference to Generative Inference: Introducing KServe v0.13
+
+We are excited to unveil KServe v0.13, marking a significant leap forward in evolving cloud native model serving to meet the demands of Generative AI inference. This release is highlighted by three pivotal updates: enhanced Huggingface runtime, robust vLLM backend support for Generative Models, and the integration of OpenAI protocol standards.
```
[the remark this answers is not in the document]
Comment (alexagriffith) [reply docs/blog/articles/2024-05-15-Kserve-0.13-release.md]: fixed

**Review by terrytangyuan — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**